hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/repository/TokenAllowanceRepository.java

```
@@ -34,4 +35,104 @@ public interface TokenAllowanceRepository extends CrudRepository<TokenAllowance,
+                        from token_transfer tt
+                            join token_allowances ta on tt.account_id = ta.owner
+                                and tt.token_id = ta.token_id,
+                                contract_results_filtered cr
+                        where tt.is_approval is true
+                            and cr.sender_id = ta.spender
+                            and tt.consensus_timestamp = cr.consensus_timestamp
+                            and tt.consensus_timestamp <= :blockTimestamp
+                            and tt.consensus_timestamp > lower(ta.timestamp_range)
+                        group by cr.sender_id, tt.token_id, tt.consensus_timestamp
+                    )
+                    select *
+                    from (
+                        select amount_granted, owner, payer_account_id, spender, timestamp_range, token_id, amount_granted + coalesce(
+                            (
+                                select amount
+                                from contract_call_transfers cct
+                                where cct.token_id = ta.token_id
+                                    and cct.sender_id = ta.spender
+                            ),
+                            0)
+                            +  coalesce(
+                            (
+                                select amount
+                                from transfers tr
+                                where tr.token_id = ta.token_id
+                                    and tr.payer_account_id = ta.spender
+                                    and tr.consensus_timestamp not in (
+                                        select consensus_timestamp
+                                        from contract_call_transfers
+                                    )
+                            ),
+                            0
+                        ) as amount
```

the `coalesce` will fail hard in real environment.

coalesce expects the individual values to be a single column from a single row.

however, here we have `contract_call_transfers` coming from a query with `group by cr.sender_id, tt.token_id, tt.consensus_timestamp`, obviously the subquery in the coalesce can return multiple rows

similarly, the second coalesce has a subquery on CTE `transfers` as a result of a query with `group by tt.token_id, tt.payer_account_id, tt.consensus_timestamp`.

In both places, `sum(amount)` is needed.

---

Comment addressed.